### PR TITLE
[Proposal] Create page.tpl and its child cms/page.tpl

### PR DESCRIPTION
**Pull request by julienbourdeau**
Here is the first proposal for the CMS page.

I created a `page.tpl` that defines most styles for a normal page: _cms_, _contact_, _store_,... And I extended it for the cms page.

I would like your opinion about:
1. Splitting into 2 templates
2. The HTML markup
3. The empty blocks (like `page_header_before`)

### CODE REVIEW

**Comment by maximebiloe** [comment]
In your page.tpl file, you may use `{block name="page_content" prepend}` and `{block name="page_content" append}`
instead of `{block name="page_content_before"}{/block}` and `{block name="page_content_after"}{/block}`

More infos about it here : http://www.smarty.net/docs/en/language.function.block.tpl

Also, maybe it would be better to set `{block name="page_content"}` inside the HTML markup to preserve it by overriding.

File: templates/cms/page.tpl [hunk 1]
```diff
@@ -0,0 +1,13 @@
+{extends "page.tpl"}
+
+{block name="content"}
+
+  {block name="page_title"}
```
Comment (maximebiloe) [inline templates/cms/page.tpl]: Shouldn't be `page_header` instead of `page_title` ?

File: templates/page.tpl [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    {block name="page_content_before"}{/block}
+    {block name="page_content"}
+      <section id="content" class="page_content">
+        <!-- Page content -->
+      </section>
+    {/block}
+    {block name="page_content_after"}{/block}
+
+    {block name="page_content_before"}{/block}
+    {block name="page_content"}
```
Comment (maximebiloe) [inline templates/page.tpl]: The block name shouldn't be `page_footer` ?

**Comment by dkarvounaris** [comment]
I see a problem with having {extends "layout.tpl"}
Think about, when a cms page is embedded as a popup somewhere, like currently in the checkout for the terms of service. It only requires the content without headers and footers then. 

Either the pages should be not implement layout.tpl directly (I think that would be the best and leave things most flexible) or layout.tpl should have features for that.

**Comment by alexsegura** [comment]
@dkarvounaris Like I said in Gitter, `layout.tpl` should be written like this: 

``` smarty
<body>
{if $content_only} 
  {block "content"}{/block}
{else} 
  // Full page content here, with columns
{/if}
</body>
```

**Comment by alexsegura** [comment]
@maximebiloe :+1: for append/prepend option, I didn't know about it

**Comment by julienbourdeau** [comment]
Thanks @maximebiloe for `prepend` and `append`. I remove the before and after blocks.

I improve a little bit the template.

Please note that this is required if you want to test it: https://github.com/PrestaShop/PrestaShop/pull/3892

**Comment by maximebiloe** [comment]
Everything looks good to me :smile:

File: templates/cms/page.tpl [hunk 3]
```diff
@@ -0,0 +1,11 @@
+{extends "page.tpl"}
+
+{block name="page_title"}
+  {$cms.meta_title}
+{/block}
+
+{block name="page_content"}
```
Comment (maximebiloe) [inline templates/cms/page.tpl]: I think the block name should be ̀page_content_container`

**Pull request merged**